src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -167,14 +168,26 @@ class IOpenGL_PhysicalDeviceBase : public IPhysicalDevice
 		std::string vendor = reinterpret_cast<const char*>(GetString(GL_VENDOR));
 		m_glfeatures.isIntelGPU = (vendor.find("Intel") != vendor.npos || vendor.find("INTEL") != vendor.npos);
 
-		float ogl_ver;
-		sscanf(reinterpret_cast<const char*>(GetString(GL_VERSION)), "%f", &ogl_ver);
-		m_glfeatures.Version = static_cast<uint16_t>(core::round(ogl_ver * 100.0f));
+		const std::regex version_re("\\s([1-9]\\.[0-9]+)");
+		std::cmatch re_match;
+
+		float ogl_ver = 0.f;
+		const char* ogl_ver_str = reinterpret_cast<const char*>(GetString(GL_VERSION));
+		if (std::regex_search(ogl_ver_str, re_match, version_re) && re_match.size() >= 2)
+		{
+			sscanf(re_match[1].str().c_str(), "%f", &ogl_ver);
+			m_glfeatures.Version = static_cast<uint16_t>(core::round(ogl_ver * 100.0f));
+		}
+		//assert(ogl_ver != 0.f);
 
-		const GLubyte* shaderVersion = GetString(GL_SHADING_LANGUAGE_VERSION);
 		float sl_ver;
-		sscanf(reinterpret_cast<const char*>(shaderVersion), "%f", &sl_ver);
-		m_glfeatures.ShaderLanguageVersion = static_cast<uint16_t>(core::round(sl_ver * 100.0f));
+		const char* shaderVersion = reinterpret_cast<const char*>(GetString(GL_SHADING_LANGUAGE_VERSION));
+		if (std::regex_search(shaderVersion, re_match, version_re) && re_match.size() >= 2)
+		{
+			sscanf(re_match[1].str().c_str(), "%f", &sl_ver);
+			m_glfeatures.ShaderLanguageVersion = static_cast<uint16_t>(core::round(sl_ver * 100.0f));
```

Comment:
careful because of versioning and semantics

If you plan to gate anything by GL version and GLSL/ESSL version then make sure any `Version>=` are paired with ES vs no ES

Reply:
? obviously, like.. theyre 2 different things